gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -1525,4 +1640,53 @@ public static AddressAggregatedListOption pageToken(String pageToken) {
    * @throws ComputeException upon failure
    */
   Operation delete(AddressId addressId, OperationOption... options);
+
+  /**
+   * Creates a new snapshot.
+   *
+   * @return a zone operation if the create request was issued correctly, {@code null} if
+   *     {@code snapshot.sourceDisk} was not found
+   * @throws ComputeException upon failure
+   */
+  Operation create(SnapshotInfo snapshot, OperationOption... options);
+
+  /**
+   * Returns the requested snapshot or {@code null} if not found.
+   *
+   * @throws ComputeException upon failure
+   */
+  Snapshot getSnapshot(String snapshot, SnapshotOption... options);
+
+  /**
+   * Lists all snapshots.
```

Comment:
Nit: I would remove "all" because options like filters, page tokens, and page size may render the returned list a subset of all snapshots.  Applies here and in ComputeRpc's list method